### PR TITLE
Add planck/scottzach1 keymap

### PR DESCRIPTION
**Pull request by scottzach1**
## Description

Adds my personal keymap for the planck keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] ~~My change requires a change to the documentation.~~
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] ~~I have added tests to cover my changes.~~
- [ ] ~~I have tested the changes and verified that they work and don't break anything (as well as I can manage).~~

### CODE REVIEW

File: keyboards/planck/keymaps/scottzach1/config.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/scottzach1/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/scottzach1/config.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/scottzach1/config.h:36]:
```suggestion
```

File: keyboards/planck/keymaps/scottzach1/config.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                                }
+#endif
+
+/*
+ * MIDI options
+ */
+
+/* Prevent use of disabled MIDI features in the keymap */
+//#define MIDI_ENABLE_STRICT 1
+
```
Comment (fauxpark) [inline keyboards/planck/keymaps/scottzach1/config.h:19]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/scottzach1/config.h (hunk 1 above)
Comment (scottzach1) [reply keyboards/planck/keymaps/scottzach1/config.h:4]: Resolved via amended commit d366e96d949041ae80c57c0469152cb68833439d

File: keyboards/planck/keymaps/scottzach1/config.h (hunk 2 above)
Comment (scottzach1) [reply keyboards/planck/keymaps/scottzach1/config.h]: Resolved via amended commit d366e96d949041ae80c57c0469152cb68833439d

File: keyboards/planck/keymaps/scottzach1/config.h (hunk 3 above)
Comment (scottzach1) [reply keyboards/planck/keymaps/scottzach1/config.h]: Resolved via amended commit d366e96d949041ae80c57c0469152cb68833439d

**Comment by scottzach1** [comment]
Thank you for your feedback. I have amended my commit and re-pushed changes. :smile:

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**